src/test/java/co/rsk/federate/signing/hsm/advanceblockchain/HsmBookkeepingClientImplTest.java

```diff
@@ -324,14 +325,19 @@ void advanceBlockchain_with_brothers_ok() throws HSMClientException, JsonRpcExce
         assertEquals(Hex.toHexString(blocks.get(1).getHeader().getFullEncoded()), blocksInRequest.get(1).asText());
         assertEquals(Hex.toHexString(blocks.get(0).getHeader().getFullEncoded()), blocksInRequest.get(2).asText());
 
-        // Assert block brothers
-        assertEquals(blocks.get(0).getUncleList().size(), brothersInRequest.get(2).size());
-        assertEquals(blocks.get(1).getUncleList().size(), brothersInRequest.get(1).size());
-        assertEquals(blocks.get(2).getUncleList().size(), brothersInRequest.get(0).size());
+        // Assert brothers
+        List<BlockHeader> block2Uncles = blocks.get(1).getUncleList();
+        String block1Brothers = brothersInRequest.get(2).toString();
+        assertTrue(block1Brothers.contains(Hex.toHexString(block2Uncles.get(0).getFullEncoded())));
+        assertTrue(block1Brothers.contains(Hex.toHexString(block2Uncles.get(1).getFullEncoded())));
```

Comment:
```suggestion
ObjectMapper objectMapper = new ObjectMapper();

        Iterator<JsonNode> payloadBrothers = objectMapper.readTree(block1Brothers).elements();
        int count = 0;
        while(payloadBrothers.hasNext()){
            String encodedBrother = payloadBrothers.next().asText();
            byte[] decode = Hex.decode(encodedBrother);
            assertArrayEquals(decode, block2Uncles.get(count).getFullEncoded());
            count++;
        }
```

Assert using `equals` to guarantee the payload was formed properly.

Reply:
Thanks Man. This Was Helpful